src/Forms/Form.php

```diff
@@ -183,6 +183,13 @@ class Form extends HtmlBuilder implements FormInterface, FieldTypeInterface
      * @var array
      */
     private $data = [];
+    
+    /**
+     * If the validator should return the value on fail or not.
+     *
+     * @var bool
+     */
+    private $validatorReturnValueOnFail = false;
```

Comment:
```suggestion

```

I know that a dedicated property and a method could be better but in order to follow existing form API, let's introduce a form option directly and leverage the `$allowedOptions` and `$defaultOptions` properties of the Form class.

Could you introduce a `flushOnFail` option in the 2 mentioned properties with a default value of `true` ? Then use the `$this->getOption('flushOnFail')` to run the condition inside the `handleRequest()` method.